aexpect/remote_door.py

```diff
@@ -341,9 +343,10 @@ def set_subcontrol_parameter(subcontrol, parameter, value):
     .. warning:: The `subcontrol` parameter is control path externally but
         control content internally after decoration.
     """
-    return re.sub(f"{parameter.upper()}[ \t\v]*=[ \t\v]*.*",
-                  f"{parameter.upper()} = {value!r}",
-                  subcontrol, count=1)
+    line = re.search(fr"{parameter.upper()}[ \t\v]*=[ \t\v]*.*", subcontrol)
+    line = "" if line is None else line.group()
+    # re.sub does undesirable post-processing of the replaced string
+    return subcontrol.replace(line, f"{parameter.upper()} = {value!r}")
```

Comment:
Hello Plamen, could you please add a selftest or at least provide the the input parameters that are failing? I tested it by:

```python
remote_door.set_subcontrol_parameter("ASD='D:\\'\nDSA = 'E:\\'", 'ASD', 'C:\\')
```

(removing the `@set_subcontrol` decorator) and the original code reported `"ASD = 'C:\\'\nDSA = 'E:\\'"` while the new version `"ASD = 'C:\\\\'\nDSA = 'E:\\'"`.

I also tried with the `@set_subcontrol` decorator and the original version worked well while this one doubled the back slashes.

Reply:
> Hello Plamen, could you please add a selftest or at least provide the the input parameters that are failing?

Have you tried the small reproducing snippet I provided in the original pull request description? Do you understand from this description what the problem I am trying to fix is? In particular:

> (removing the `@set_subcontrol` decorator) and the original code reported `"ASD = 'C:\\'\nDSA = 'E:\\'"` while the new version `"ASD = 'C:\\\\'\nDSA = 'E:\\'"`.

How did you test the "reported output" here? Was it via `print()` (analogous to the `@set_subcontrol` decorator's writing the string to a file) or perhaps just an interpreter's repr output like
```
>>> reported_string = <some-manupulaton>
>>> reported_string
```
If the string that these functions output is written to a file or printed you will see:
```
>>> import re
>>> re.sub("ASD = 'C:\\'\nDSA = 'E:\\'", "ASD = 'C:\\'\nDSA = 'E:\\'", 'C:\\')
'C:\\'
>>> print(re.sub("ASD = 'C:\\'\nDSA = 'E:\\'", "ASD = 'C:\\'\nDSA = 'E:\\'", 'C:\\'))
C:\
```
The second output when written into a file from a metaprogramming point of view will result in a syntax error.

Reply:
I used interactive interpreter but print/write works differently, my fault. So yes, this works definitely better and thank you for the confirmation.

Reply:
No worries